File: conexiones/conexionMongo.go
package main

import (
	"context"
	"fmt"
	"time"

	"go.mongodb.org/mongo-driver/mongo"
	"go.mongodb.org/mongo-driver/mongo/options"
)

func main() {
	clientOptions := options.Client().ApplyURI("mongodb://localhost:27017")
	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
	defer cancel()

	client, err := mongo.Connect(ctx, clientOptions)
	if err != nil {
		fmt.Println("Error al conectar:", err)
		return
	}

	err = client.Ping(ctx, nil)
	if err != nil {
		fmt.Println("Error al hacer ping a MongoDB:", err)
		return
	}
	fmt.Println("Conexión exitosa a MongoDB")
	// client.Disconnect(ctx) // Desconecta si es necesario
}

File: conexiones/conexionMongo.js
const { MongoClient } = require('mongodb');

async function run() {
    const client = new MongoClient("mongodb://localhost:27017/");
    try {
        console.log("Conexión exitosa");
        await client.connect();
    } catch (err) {
        console.error(err);
    } finally {
        await client.close();
    }
}

run();
